src/ar_storage.erl

```diff
@@ -59,19 +60,20 @@ block_exists(Hash) ->
 %% @doc Move a block into the 'invalid' block directory.
 invalidate_block(B) ->
 	ar_block_index:remove(B#block.indep_hash),
+	DataDir = ar_meta_db:get(data_dir) ++ "/",
 	TargetFile =
 		lists:flatten(
 			io_lib:format(
 				"~s/invalid/~w_~s.json",
-				[?BLOCK_DIR, B#block.height, ar_util:encode(B#block.indep_hash)]
+				[DataDir ++ ?BLOCK_DIR, B#block.height, ar_util:encode(B#block.indep_hash)]
```

Comment:
Let's put the whole `lists:flatten/1` call in a function that takes a block as argument.

Reply:
✅